### ISSUE TITLE
[BUG] Straight should not conclude these two version strings indicate a different Emacs Version.

### ISSUE THREAD

**Issue by jasonm23** [radian-software/straight.el#995]
## What's wrong

Straight is checking `(emacs-version)` and using it, without sanitizing, to determine the need to build again.

For example:

```lisp
(emacs-version)

=> "GNU Emacs 28.1 (build 1, x86_64-apple-darwin20.6.0, Carbon Version 164 AppKit 2202.7)
 of 2022-08-02"
```
Which is effectively the same as a build done the next day with the same code.
```lisp
(emacs-version)

=> "GNU Emacs 28.1 (build 1, x86_64-apple-darwin20.6.0, Carbon Version 164 AppKit 2202.7)
 of 2022-08-03"
```

A sanitized version check could avoid unnecessary errors, for example:

```lisp
;; draft implementation
(defun straight-emacs-version () 
  "Emacs version sanitized."
  (let ((version-string (emacs-version)))
     (replace-regexp-in-string "^\\(.*\\)\n.*" "\\1" version-string)  
```

**Comment by progfolio** [comment]
Agreed. Even better would probably be to jus use the emacs-version variable.

**Comment by jasonm23** [comment]
What's worse is that a lot of functionality that shouldn't be blocked by this, e.g. `(describe-function straight-use-package)` ... is.  It's infuriating, given the total build time to "fix" it.

**Comment by jasonm23** [comment]
Although a quick way to fix it is just to `(byte-compile-file "~/.emacs.d/.local/straight/build-nn.n/straight/straight.el")

I'm referring to the Doom Emacs "kill .local & rebuild" style _fix_.

**Comment by raxod502** [comment]
Eh? We changed from using `(emacs-version)` to just `emacs-version` in the build cache here: https://github.com/radian-software/straight.el/commit/cb7679cee0ea05b589607de9080e9d1b144c92bb.

I see that we are still using `(emacs-version)` here:

https://github.com/radian-software/straight.el/blob/772a47396ace9381c627307709bfaa6072bec751/straight.el#L42-L49

BUT, that is only for the byte-compiling of `straight.el` itself, which should take at most a second, it will not cause the rest of the packages to be rebuilt.

Anyway, it's a simple improvement to use the same variable for this as for the build cache, so I will do in any case.
